workshop/posting-error-example.md

```diff
@@ -0,0 +1,201 @@
+
+<img src = "screenshots/chunking_code.png" width = "500">
+
+- Go back to the beginning of your notebook and re-run your code in order and make sure you haven't missed any vital steps! Order matters!
+
+## Step 2) Draft out your question you'll post.
+
+You may want to open up a text editor to write this out before posting (this is a personal preference thing).
+
+For our example error, here's how each of the "four points" (and the optional one) could be included in your posted question; we will go through the details on each point:
+
+<img src = "screenshots/breakdown.png">
+
+### Where is this error occurring; what notebook/chunk?
+
+- In our workshop, notebook references tend to make sense to provide, however, more generally this should be any context around __where__ you've encountered this error.
+
+- If you are *not* working from any CCDL materials, it is *even more important* to provide context around what the notebook you are creating is supposed to be doing.
+
+- Note that line numbers change as you've added code, so line numbers as a reference aren't as helpful.
+
+*For our example:*  
+<img src = "screenshots/where-is-the-error.png" width = "600">
+
+### What is the goal of this code?  
+
+- Describing your end goals for this code will provide context for others so they can tailor their advice with your goal in mind.
+    *Examples of information you may want to include (whenever it is applicable):*
+    - What is your scientific question?
+    - What does your input data look like?
+       - What kind of file *e.g.* `.TSV`, `.fastq`?
+       - What information is included the data?
+       - What are the columns?
+       - What are the rows?
+    - What'd your desired output?
+       - What kind of results are you looking for e.g. a plot, a table, processed/filtered data?
+       - What kind of file *e.g.* `.TSV`, `.fastq`?
+
+- Sometimes you can bypass an error completely by using an alternative strategy altogether, but your helper won't be able to suggest an alternative if they don't know your ultimate goal.
+
+*For our example:*  
+<img src = "screenshots/what-is-the-goal.png" width = "600">
+
+### What is the code that is producing this error?
+
+- You should [use backticks (```) to format your code.](../virtual-setup/slack-procedures.md#adding-code-blocks-to-messages) (How to do this in the next step).
+Formatting your code makes it easier for others to distinguish between your code and the words in your post where you are otherwise describing the problem.
+
+- Screenshots are not ideal because others can't copy-paste your code from a screenshot. Being able to copy-paste the code with backticks makes it easiest for others to help you.
+
+*For our example:*  
+<img src = "screenshots/what-is-the-code.png" width = "600">
+
+### What is the error or problematic outcome?  
+
+- Include any error messages verbatim. Also preferably not a screenshot so others can copy-paste for Google searching.
```

Comment:
Am I being too snarky with this parenthetical? Maybe, but if we are going to give away our secrets, let's encourage people to apply them to their own work?

```suggestion
- Include any error messages verbatim. 
Also preferably not a screenshot so others can copy-paste for Google searching (Did you try that? It's often surprisingly helpful!).
```

Reply:
I'll probably change the wording to de-snark a tad, but I'll keep the sentiment. :+1: